## Verification — green CI does not prove this change works

The tree has no dead code, so the step produces no output and a green Lint job says nothing about whether the blocking was actually removed. Both the reporting path and the non-blocking behaviour need to be driven with findings present.

So I extracted the **real** `run:` block out of `lint.yml` with a YAML parser — not a retyped copy — stubbed `go install` / `go env GOPATH`, put a fake `deadcode` on that GOPATH that emits findings, and ran it:

```
::warning::Dead code detected (reported, not blocking — see #3014):
daemon/sandbox_token.go:54:6: unreachable func: (*sandboxTokenRegistry).mint
daemon/sandbox_token.go:95:6: unreachable func: (*sandboxTokenRegistry).revoke
  -> exit=0
```

Job summary, verbatim:

> ### Dead code — reported, not blocking
>
> These symbols are unreachable. That is usually a defect worth fixing; it is not a merge blocker, because a feature may legitimately land its mechanism before its call site (#3014). Read the list before assuming it is staged.
>
> ```
> daemon/sandbox_token.go:54:6: unreachable func: (*sandboxTokenRegistry).mint
> daemon/sandbox_token.go:95:6: unreachable func: (*sandboxTokenRegistry).revoke
> ```

Exit 0 with the findings still named, in both places.

**Worth recording: my first attempt at this verification retyped the step body by hand and had a quoting bug** — `echo "```"` inside a double-quoted context is command substitution, not a literal fence — which silently swallowed the findings list out of the summary while still exiting 0. It would have "passed". The workflow itself uses `'```'` and is fine, but that is exactly why the check above runs the extracted block rather than an approximation of it: a hand-copied harness can pass while the real thing is broken, or fail while the real thing is fine, and neither tells you anything.

The same extraction confirms `pr.yml` carries an identical step; both files re-parse as valid YAML with the step named `Report dead code`, and `exit 1` is gone from that step in both.

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -61,7 +61,17 @@ jobs:
           go install golang.org/x/tools/cmd/deadcode@v0.48.0
           out=$("$(go env GOPATH)/bin/deadcode" -test ./...)
           if [ -n "$out" ]; then
-            echo "::warning::Dead code detected (reported, not blocking — see #3014):"
+            # A workflow command is ONE line: a bare `echo "$out"` after it is an
+            # ordinary log record, not part of the annotation, so the annotation
+            # would read "Dead code detected" and name nothing. Newlines must be
+            # %0A-escaped into the command itself. Escape % FIRST or the escapes
+            # inserted below get double-escaped.
+            encoded=$(printf '%s' "$out" \
+              | sed -e 's/%/%25/g' -e 's/\r/%0D/g' \
+              | awk 'BEGIN{ORS=""} {print (NR>1 ? "%0A" : "") $0}')
+            echo "::warning::Dead code detected (reported, not blocking — see #3014):%0A${encoded}"
+            # Kept as plain log output too: the annotation is for the reviewer, this
+            # is what stays greppable in the raw log.
             echo "$out"
             {
               echo "### Dead code — reported, not blocking"

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -76,7 +76,17 @@ jobs:
           go install golang.org/x/tools/cmd/deadcode@v0.48.0
           out=$("$(go env GOPATH)/bin/deadcode" -test ./...)
           if [ -n "$out" ]; then
-            echo "::warning::Dead code detected (reported, not blocking — see #3014):"
+            # A workflow command is ONE line: a bare `echo "$out"` after it is an
+            # ordinary log record, not part of the annotation, so the annotation
+            # would read "Dead code detected" and name nothing. Newlines must be
+            # %0A-escaped into the command itself. Escape % FIRST or the escapes
+            # inserted below get double-escaped.
+            encoded=$(printf '%s' "$out" \
+              | sed -e 's/%/%25/g' -e 's/\r/%0D/g' \
+              | awk 'BEGIN{ORS=""} {print (NR>1 ? "%0A" : "") $0}')
+            echo "::warning::Dead code detected (reported, not blocking — see #3014):%0A${encoded}"
+            # Kept as plain log output too: the annotation is for the reviewer, this
+            # is what stays greppable in the raw log.
             echo "$out"
             {
               echo "### Dead code — reported, not blocking"

